apis/cloud-api-manager/src/main/services/tyk_dashboard.js

```diff
@@ -0,0 +1,224 @@
+// jshint esversion:6
+const axios = require('axios').default;
+const logger = new (require('../logging/custom_logger'))("cloud-api-manager:services:tyk_dashboard");
+
+class TykDashboardService {
+
+    /**
+     * Creates an instance of the TykDashboardService
+     * @param {string} authorization 
+     * @param {string} baseUrl 
+     */
+    constructor(authorization, baseUrl = 'https://admin.cloud.tyk.io') {
+        this._httpClientInstance = axios.create({
+            baseURL: baseUrl,
+            timeout: 1000,
+            headers: { 'authorization': authorization },
+            validateStatus: function (status) {
+                return status >= 200 && status < 300; // default
```

Comment:
Yeah, in javascript, functions can be passed around as arguments. In this case, this function will be evaluated once the http response comes through and if its status code is outside those bounds, it will get rejected.